### DISCUSSION TITLE
[KONG] - Skywalking Kong agent parameters - more info

### DISCUSSION THREAD

**Discussion by andrewgkew** [apache/skywalking#7998] [Q&A]
Can I get get more of an explanation as to the parameters I can configure for the Kong agent

1. backend_http_uri
2. service_name
3. service_instance_name

I am assuming the backend http uri is the OAP server? But the example is using port 12800?
Also what the service name and instance name are so that I know in my use case what they should be set to

Thanks

**Comment by wu-sheng** [comment]
> I am assuming the backend http uri is the OAP server? But the example is using port 12800?

Yes, it is 12800. Because there is no gRPC core in Nginx stack.

Service name and instance name are basically logic names, you could find the explanations here, https://skywalking.apache.org/docs/main/v8.8.1/en/concepts-and-designs/overview/#why-use-skywalking

**Reply by andrewgkew** [reply]
Thanks @wu-sheng 

So the backend is still the OAP server and it will be listening on port 12800? so its 11800 for gRPC and 12800 for HTTP?
Do I need to adjust the OAP configuration for this to work?

Reading those docs (thanks BTW) makes sense as logic names and I assume those are what you see in the UI when viewing metrics in each?

I looked at the Lua code of the agent (I am familiar with Kong custom plugins) and I see you are using some of your own custom logic with regards to the tracing, but if OAP has a HTTP endpoint, would I be able to use the Kong Zipkin plugin setting the OAP server at 12800 and using OpenTelemetry headers to get the same behaviour as this plugin? Not sure if you have looked into this before?

**Reply by wu-sheng** [reply]
> So the backend is still the OAP server and it will be listening on port 12800? so its 11800 for gRPC and 12800 for HTTP?

Yes

> Do I need to adjust the OAP configuration for this to work?

No, it is in default.

> but if OAP has a HTTP endpoint, would I be able to use the Kong Zipkin plugin setting the OAP server at 12800 and using OpenTelemetry headers to get the same behaviour as this plugin? Not sure if you have looked into this before?

Zipkin HTTP receiver is always there for a long time. But Zipkin trace(or similar, like Jaeger) is not good at analyzing topology, metrics and alarm. More importantly, it doesn't work with other SkyWalking's agent. The header is incompatible. 
Recommend to use native tracers. The SkyWalking uses our own protocols, https://skywalking.apache.org/docs/main/v8.8.1/en/protocols/readme/#language-based-native-agent-protocol.

APM system design is far away from pure tracing in Zipkin and OpenTelemetry. So, we have all in our own implementation.

**Reply by andrewgkew** [reply]
Thanks @wu-sheng I am going to put this together and hopefully use the Kong plugin with SkyWalking at another conference next week (Devoxx UK) this time in person but I think its virtual and in person too. Same topic as Kong Summit but just adding another dimension with the gateway in the spans.

**Reply by andrewgkew** [reply]
Sorry got that wrong here is what we are talking at is a MeetUp at Devoxx: https://www.meetup.com/Kong-LONDON/events/281516512/

**Reply by wu-sheng** [reply]
Good to see a new presentation about this. Look forward to your recorded video too.

**Reply by andrewgkew** [reply]
Just a quick one, got the plugin working on EKS cluster :)
Bit of a weird way to install it though (working with Kong engineers to get your 3rd party dependency added)
I might put a PR through for your README.md on how to install the plugin using helm when Kong is deployed with helm on k8s
But thanks for your help, can see kong now in my topology and in my traces 👍

**Reply by wu-sheng** [reply]
Good to know. If something missed in the doc, please send a pull request to fix.
